### DISCUSSION TITLE
Remote files editing in Helix via SSH ?

### DISCUSSION THREAD

**Discussion by pm-hwks** [helix-editor/helix#2219] [General]
Does helix support editing remote files via SSH ?
VSCode allows this via remote development plugins and emacs provides this functionality via tramp

**Comment by David-Else** [comment]
Have you tried a recent version of Kitty terminal? You should be able to run Helix in that and edit remote files using the `ssh kitten` that is built in: https://sw.kovidgoyal.net/kitty/faq/#i-get-errors-about-the-terminal-being-unknown-or-opening-the-terminal-failing-when-sshing-into-a-different-computer

https://sw.kovidgoyal.net/kitty/changelog/#truly-convenient-ssh

> The [ssh kitten](https://sw.kovidgoyal.net/kitty/kittens/ssh/) is redesigned with powerful new features:
>   Automatic [Shell integration](https://sw.kovidgoyal.net/kitty/shell-integration/#shell-integration) on remote machines
>  Easily [clone local shell/editor config](https://sw.kovidgoyal.net/kitty/kittens/ssh/#real-world-ssh-kitten-config) on remote machines
 > Automatic [re-use of existing connections](https://sw.kovidgoyal.net/kitty/kittens/ssh/#opt-kitten-ssh.share_connections) to avoid connection setup latency

**Reply by kshitijsachan** [reply]
Just to make sure I'm understanding, this won't solve the problem of latency from key presses or filepicker search, right?

**Reply by jmg-duarte** [reply]
I don't understand this reply, where did OP say that he was using kitty? 

If I understand correctly, your suggestion is that anyone who wants to get `hx` working over SSH, moves to a specific terminal?

**Comment by vvzen** [comment]
As a remote worker for the last 2 years+, I have been using helix via ssh in a more basic way: I compiled helix on the target machine(s) where I constantly ssh, installed `rust-anaylizer` there too, and then started coding. 
Performance-wise, it's been way better than using `sshfs` or other options, the only annoying thing is support for copy pasting (space+p)from the local machine that I'm using, instead of the one that I'm ssh-ing to.

Does anyone know of a way to achieve that ? 
Using the classic cmd/ctrl+c cmd/ctrl + v is really laggy over big chunks of texts, and also formats it incorrectly.

**Reply by zacharyburnett** [reply]
does `:clipboard-yank` work?

EDIT: oh this comment is 2 years old, you probably figured that out already 😅

**Comment by NewtonChutney** [comment]
I'm working in an org that has very old builds of everything.. rhel6, centos7 are the most probable oses you'll land on in a server..

I'd like to be able to ssh from a Helix instance running in a local terminal.. Compiling/installing on the remote envs are out of scope.. 🥲

**Reply by NewtonChutney** [reply]
And as the local glibc versions are quite different from the remote.. I don't think Helix can handle sending over a server binary, to assist with remote directory traversing and file editing.

**Comment by nullbyto** [comment]
It would be a great feature, to add support for editing files on a remote server using the `scp` protocol like in `vim`:
```bash
vim scp://user@myserver[:port]//path/to/file
```

**Comment by SHU-red** [comment]
+1

One Setup to rule them all

**Comment by abu-matterize** [comment]
Do-able in 2024?

**Reply by markstos** [reply]
Not directly that I'm aware. You could use something like `sshfs` to have the OS mount a folder over SSH, such that Helix would see the remote files as being local.
